localstack/services/events/events_starter.py

```diff
@@ -198,3 +232,119 @@ def start_events(port=None, asynchronous=None, update_listener=None):
+                elif not element_value and event_value:
+                    evaluated_result.append(False)
+                else:
+                    evaluated_result.append(True)
+            elif element_key.lower() == 'cidr':
+                ips = [str(ip) for ip in ipaddress.IPv4Network(element_value)]
+                if event_value not in ips:
+                    evaluated_result.append(False)
+                else:
+                    evaluated_result.append(True)
+            elif element_key.lower() == 'numeric':
+                list_of_operators = element_value
+                check_valid_numeric_content_base_rule(list_of_operators)
+                for index in range(len(list_of_operators)):
+
+                    if list_of_operators[index] == '>' and \
+                       isinstance(list_of_operators[index + 1], int) and \
+                       event_value <= list_of_operators[index + 1]:
+                        evaluated_result.append(False)
+
+                    elif list_of_operators[index] == '>=' and \
+                        isinstance(list_of_operators[index + 1], int) and \
+                            event_value < list_of_operators[index + 1]:
+                        evaluated_result.append(False)
+
+                    elif list_of_operators[index] == '<' and \
+                        isinstance(list_of_operators[index + 1], int) and \
+                            event_value >= list_of_operators[index + 1]:
+                        evaluated_result.append(False)
+
+                    elif list_of_operators[index] == '<=' and \
+                        isinstance(list_of_operators[index + 1], int) and \
+                            event_value > list_of_operators[index + 1]:
+                        evaluated_result.append(False)
+
+                    else:
+                        evaluated_result.append(True)
+                    index = index + 1
+
+            elif element_key.lower() == 'anything-but':
+                if isinstance(element_value, list) and \
+                   event_value in element_value:
+                    evaluated_result.append(False)
+                elif (isinstance(element_value, str) or isinstance(element_value, int)) and \
+                        event_value == element_value:
+                    evaluated_result.append(False)
+                elif isinstance(element_value, dict):
+                    nested_key = list(element_value)[0]
+                    if nested_key == 'prefix' and re.match(r'^{}'.format(element_value.get(nested_key)), event_value):
+                        evaluated_result.append(False)
+                    else:
+                        evaluated_result.append(True)
+                else:
+                    evaluated_result.append(True)
+            else:
+                evaluated_result.append(True)
+    if any(evaluated_result) or evaluated_result == []:
+        return True
+    elif not all(evaluated_result):
```

Comment:
This `elif` branch is redundant - if it matches, then we return `False`, if it does not match, then the function returns `None`. Both values are interpreted as _falsy_ values when evaluated in line 77 further above.

What is the expected behavior here - shouldn't this function return `True` only if _all_ values in `evaluated_result` are true?

(I think this function should also be more thoroughly unit tested...)